sormas-backend/src/main/java/de/symeda/sormas/backend/campaign/statistics/CampaignStatisticsService.java

```diff
@@ -0,0 +1,403 @@
+import java.util.stream.Stream;
+
+import javax.ejb.LocalBean;
+import javax.ejb.Stateless;
+import javax.persistence.EntityManager;
+import javax.persistence.PersistenceContext;
+import javax.persistence.Query;
+
+import de.symeda.sormas.api.campaign.CampaignJurisdictionLevel;
+import de.symeda.sormas.api.campaign.data.CampaignFormDataEntry;
+import de.symeda.sormas.api.campaign.form.CampaignFormElement;
+import de.symeda.sormas.api.campaign.form.CampaignFormElementType;
+import de.symeda.sormas.api.campaign.statistics.CampaignStatisticsCriteria;
+import de.symeda.sormas.api.campaign.statistics.CampaignStatisticsDto;
+import de.symeda.sormas.api.campaign.statistics.CampaignStatisticsGroupingDto;
+import de.symeda.sormas.api.utils.SortProperty;
+import de.symeda.sormas.backend.campaign.Campaign;
+import de.symeda.sormas.backend.campaign.data.CampaignFormData;
+import de.symeda.sormas.backend.campaign.form.CampaignFormMeta;
+import de.symeda.sormas.backend.region.Area;
+import de.symeda.sormas.backend.region.Community;
+import de.symeda.sormas.backend.region.District;
+import de.symeda.sormas.backend.region.Region;
+import de.symeda.sormas.backend.util.ModelConstants;
+
+@Stateless
+@LocalBean
+public class CampaignStatisticsService {
+
+	@PersistenceContext(unitName = ModelConstants.PERSISTENCE_UNIT_NAME)
+	private EntityManager em;
+
+	public List<CampaignStatisticsDto> getCampaignStatistics(
+		CampaignStatisticsCriteria criteria,
+		Integer first,
+		Integer max,
+		List<SortProperty> sortProperties) {
+
+		Query campaignsStatisticsQuery = em.createNativeQuery(buildStatisticsQuery(criteria, sortProperties));
+		final CampaignJurisdictionLevel groupingLevel = criteria.getGroupingLevel();
+		Map<CampaignStatisticsGroupingDto, CampaignStatisticsDto> results = new LinkedHashMap<>();
+		((Stream<Object[]>) campaignsStatisticsQuery.getResultStream()).forEach(result -> {
+			CampaignStatisticsGroupingDto campaignStatisticsGroupingDto = new CampaignStatisticsGroupingDto(
+				(String) result[1],
+				(String) result[2],
+				(String) result[3],
+				shouldIncludeRegion(groupingLevel) ? (String) result[4] : "",
+				shouldIncludeDistrict(groupingLevel) ? (String) result[5] : "",
+				shouldIncludeCommunity(groupingLevel) ? (String) result[6] : "");
+			if (!results.containsKey(campaignStatisticsGroupingDto)) {
+				CampaignStatisticsDto campaignStatisticsDto =
+					new CampaignStatisticsDto(campaignStatisticsGroupingDto, result[0] != null ? ((Number) result[0]).intValue() : null);
+				results.put(campaignStatisticsGroupingDto, campaignStatisticsDto);
+			}
+			int length = result.length;
+			CampaignFormDataEntry campaignFormDataEntry = new CampaignFormDataEntry((String) result[length - 2], result[length - 1]);
+			results.get(campaignStatisticsGroupingDto).addStatisticsData(campaignFormDataEntry);
+		});
+		return results.values().stream().collect(Collectors.toList());
+	}
+
+	public long count(CampaignStatisticsCriteria criteria) {
```

Comment:
I guess the count query is useless right now because we don't have lazy loading, so please comment it out for now